src/cgame/default/cg_hud.c

```diff
@@ -1107,12 +1144,29 @@ static void Cg_DrawSelectWeapon(const player_state_t *ps) {
 	r_pixel_t ch;
 	cgi.BindFont("medium", NULL, &ch);
 
+	if (cg_select_weapon_fade->modified || cg_select_weapon_interval->modified) {
+		cg_select_weapon_fade->modified = false;
+
+		cg_select_weapon_fade->value = Clamp(cg_select_weapon_fade->value, 0.0, cg_select_weapon_interval->value);
+	}
```

Comment:
Fade is a good idea for this, I need to test it to see how it feels though. If I was to implement fade it would be an additional amount of seconds added to the display interval probably.

Reply:
What does this do? Blend the weapon during the switch animation?

Reply:
No, it fades out the weapon bar a second or so after switching weapons.